src/common/transformations/src/transformations/smart_reshape/lstm_states_broadcast.cpp

```diff
@@ -0,0 +1,181 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <memory>
+#include <ngraph/pass/manager.hpp>
+#include <openvino/op/util/sub_graph_base.hpp>
+#include <openvino/opsets/opset9.hpp>
+#include <transformations/smart_reshape/lstm_states_broadcast.hpp>
+#include <transformations/utils/utils.hpp>
+
+#include "dimension_tracker.hpp"
+#include "itt.hpp"
+
+ov::Input<ov::Node> get_outer_input_of_ti_by_parameter(const std::shared_ptr<ov::opset9::Parameter>& parameter,
+                                                       const std::shared_ptr<ov::opset9::TensorIterator>& ti) {
+    const auto& body = ti->get_body();
+    OPENVINO_ASSERT(body != nullptr, "TI returns invalid body graph ", ti);
+    int64_t parameter_index = ti->get_body()->get_parameter_index(parameter);
+    OPENVINO_ASSERT(parameter_index >= 0,
+                    "LSTMStatesBroadcast encountered unregistered parameter ",
+                    parameter,
+                    " related to TI body ",
+                    ti);
+    for (const auto& input_descriptor : ti->get_input_descriptions())
+        if (input_descriptor->m_body_parameter_index == parameter_index)
+            return ti->input(input_descriptor->m_input_index);
+    OPENVINO_UNREACHABLE("LSTMStatesBroadcast failed to get outer input of TI by its inner Parameter. TI ",
+                         ti,
+                         " Parameter ",
+                         parameter);
+}
+
+std::shared_ptr<ov::Node> deduce_outer_source_of_batch_for_inner_lstm_cell(
+    const std::shared_ptr<ov::opset9::TensorIterator>& ti,
```

Comment:
I would recommend to use:
`using namespace ov::opset9;
 using namespace std;
 `
 and so on in *.cpp files to simplify code.